java/compaction-core/src/main/java/sleeper/compaction/job/CompactionJob.java

```diff
@@ -40,6 +40,7 @@ public class CompactionJob {
     private int dimension; // Determines the row key to be used for splitting
     private String iteratorClassName;
     private String iteratorConfig;
+    private String taskId;
```

Comment:
We don't really need to put the task ID on this class. It can only exist after the compaction job is pulled off the queue by a task. It might be clearer to leave it out.